projects/ng-date-picker/src/lib/calendar/calendar.component.ts

```diff
@@ -322,7 +329,18 @@ export class CalendarComponent implements OnInit, AfterViewInit {
   private initSecondCalendar(): void {
     const currDate = new Date();
     this.secondCalendarViewData = new CalendarViewData();
-    this.secondCalendarViewData.minDate = this.getFirstDateOfNextMonth(this._maxDate.cal1);
+
+    if (this.selectedDates.end && this.selectedDates.start 
+      && this.selectedDates.end.getMonth() === this.selectedDates.start.getMonth()
+      && this.selectedDates.end.getFullYear() === this.selectedDates.start.getFullYear()
+      && (this.selectedDates.end.getMonth() !== 11 
+      || (this.selectedDates.end.getFullYear() !== currDate.getFullYear() && this.selectedDates.end.getMonth() === 11))
+    ) { // range for same month
+      this.secondCalendarViewData.minDate = this.getFirstDateOfNextMonth(this.selectedDates.end);
+    } else {
+      this.secondCalendarViewData.minDate = this.getSecondeCalendarMinDate(currDate);
+    }
```

Comment:
why do you need all this logic? 🤔 the logic before was "first day of the next month of the maxDate on the calendar 1 (which is the start date)".. what was the issue on that?